### ISSUE TITLE
Trade between players can result in loss if item(s)

### ISSUE THREAD

**Issue by Wiggo32** [DarkstarProject/darkstar#4563]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:** 30171223_0

**_Server Version_** (type `!revision` in game) **:** fae4c89

**_Source Branch_** (master/stable) **:** master

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 

1. Grab a buddy
2. Load up your inventory so you have no slots left
3. Have your buddy trade you an item.
4. Note the error message "You cannot trade" due to you having no spots available for the trade
5. Have your buddy trade you the same item but this time, include an item of your own in the trade window.
6. Note the "trade complete" message.
7. Also, note that your item was never received from your buddy and you now have 1 spot available.
  --the item just disappears into the nothingness never to be seen again
8. Also, note the if both players inventories are completely full and a trade happens between them, only 1 of the items traded will disappear into the void.

**Comment by teschnei** [comment]
I'm guessing on retail both parties must have enough space before the trade
starts

**Comment by Wiggo32** [comment]
That is correct, and gil counts as an item. So, even with 1 slot (and i just did this on retail) you cannot accept an item plus gil.

**Comment by Hozu** [comment]
@Wiggo32 Huh, I wonder if that's why there's an item ID for gil.

**Comment by kaincenteno** [comment]
Experienced this issue on Version: 30180329_1
Same steps to reproduce behavior, item lost.

**Comment by kaincenteno** [comment]
I havent experienced this lately, cannot say if it was fixed or not

**Comment by TeoTwawki** [comment]
so this is still a thing, just reminding. 

> [7:03 PM] Rheine: just traded two merman rings from one toon to another vs 3 stacks of crystals. bad trade, i know. toon to receive rings was 80/80 thats why i switched the crystals, to make room. trade completed. but the rings have disapeared. lost. just a glitch or me bad from trying that trade?

Server log shows:

> [0m[01/Dec] [15:55:31][1;37m[Notice][0m [1;36mRheine->Cat trade updating trade slot id 1 with item mermans_ring, quantity 1
[0m[01/Dec] [15:55:32][1;37m[Notice][0m [1;36mRheine->Cat trade updating trade slot id 2 with item mermans_ring, quantity 1
[0m[01/Dec] [15:55:53][1;37m[Notice][0m [1;36mRheine->Cat trade updating trade slot id 1 with item mermans_ring, quantity 1
[0m[01/Dec] [15:55:54][1;37m[Notice][0m [1;36mRheine->Cat trade updating trade slot id 2 with item mermans_ring, quantity 1
[0m[01/Dec] [15:55:57][1;37m[Notice][0m [1;36mCat->Rheine trade updating trade slot id 1 with item wind_crystal, quantity 12
[0m[01/Dec] [15:55:59][1;37m[Notice][0m [1;36mCat->Rheine trade updating trade slot id 2 with item wind_crystal, quantity 12
[0m[01/Dec] [15:56:01][1;37m[Notice][0m [1;36mCat->Rheine trade updating trade slot id 3 with item wind_crystal, quantity 12

I queried inventories and verified neither player has those merman's rings.
